### PR TITLE
Docs: Document which data sources work with caching

### PR DESCRIPTION
**Pull request by mjseaman**
**What this PR does / why we need it**:
Adds some extra docs to clarify which built-in and Enterprise data sources work with query caching, and how to tell if other third-party data sources work with query caching.

**Which issue(s) this PR fixes**:
None- this fixes an issue encountered by a user: https://raintank-corp.slack.com/archives/CNS0TK5BN/p1625613824197500

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/query-caching.md [hunk 1]
```diff
@@ -27,6 +27,13 @@ You can make a panel retrieve fresh data more frequently by increasing the **Max
 - Reduced API costs.
 - Reduced likelihood that APIs will rate-limit or throttle requests.
 
+## Data sources that work with query caching
+Query caching works for all [Enterprise data sources](https://grafana.com/grafana/plugins/?type=datasource&enterprise=1), and it works for all [built-in data sources]({{< relref "../datasources/_index.md" >}}) except for Elasticsearch, Prometheus, and Loki, which cache queries themselves.
```
Comment (kminehart) [inline docs/sources/enterprise/query-caching.md:31]: We probably need to verify this: `it works for all [built-in data sources]({{< relref "../datasources/_index.md" >}}) except for Elasticsearch, Prometheus, and Loki, which cache queries themselves`.

File: docs/sources/enterprise/query-caching.md [hunk 2]
```diff
@@ -27,6 +27,22 @@ You can make a panel retrieve fresh data more frequently by increasing the **Max
 - Reduced API costs.
 - Reduced likelihood that APIs will rate-limit or throttle requests.
 
+## Data sources that work with query caching
```
Comment (oddlittlebird) [inline docs/sources/enterprise/query-caching.md:30]:
```suggestion
## Data sources that work with query caching

```

File: docs/sources/enterprise/query-caching.md [hunk 3]
```diff
@@ -27,6 +27,22 @@ You can make a panel retrieve fresh data more frequently by increasing the **Max
...
+Query caching works for all [Enterprise data sources](https://grafana.com/grafana/plugins/?type=datasource&enterprise=1), and it works for the following [built-in data sources]({{< relref "../datasources/_index.md" >}}):
+- InfluxDB
```
Comment (oddlittlebird) [inline docs/sources/enterprise/query-caching.md:32]: Alphabetize this list, please.

File: docs/sources/enterprise/query-caching.md [hunk 4]
```diff
@@ -27,6 +27,22 @@ You can make a panel retrieve fresh data more frequently by increasing the **Max
...
+- Tempo
+- MySQL
+- Postgres
+- Microsoft SQL Server
+- Google Cloud Monitoring
+- CloudWatch
+
+Some data sources,like  Elasticsearch, Prometheus, and Loki, cache queries themselves and therefore Grafana query caching does not improve performance.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/query-caching.md:40]:
```suggestion
Some data sources, such as Elasticsearch, Prometheus, and Loki, cache queries themselves, so Grafana query caching does not improve performance.
```

File: docs/sources/enterprise/query-caching.md [hunk 5]
```diff
@@ -27,6 +27,22 @@ You can make a panel retrieve fresh data more frequently by increasing the **Max
...
+
+Query caching also works for all data sources that include a backend. More specifically, caching works with data sources that extend the `DataSourceWithBackend` class in the plugins SDK.
+
+To tell if a data source works with query caching, follow the instructions below to **Enable and Configure query caching**. If Caching is enabled in Grafana but the Caching tab is not visible for the given data source, then query caching is not available for that data source.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/query-caching.md:44]:
```suggestion
To tell if a data source works with query caching, follow the instructions below to **Enable and Configure query caching**. If caching is enabled in Grafana but the Caching tab is not visible for the given data source, then query caching is not available for that data source.
```

File: docs/sources/enterprise/query-caching.md (hunk 1 above)
Comment (mjseaman) [reply docs/sources/enterprise/query-caching.md]: Went through and confirmed which built-in data sources display the Cache tab and which ones don't.

File: docs/sources/enterprise/query-caching.md (hunk 3 above)
Comment (mjseaman) [reply docs/sources/enterprise/query-caching.md:35]: Done :)

**Review by oddlittlebird — APPROVED** [review]
Looks good!

**Pull request merged**